db/seeds/scheme_ten.rb

```diff
@@ -50,18 +53,46 @@
 csv_file = File.open(file_path, 'r:ISO-8859-1')
 csv = CSV.parse(csv_file, headers: true)
 
+module OffenceCSVRowExtensions
+  def description
+    self['description'].strip
+  end
+
+  def offence_category
+    OffenceCategory.find_by(number: self['category'])
+  end
+
+  def offence_band
+    OffenceBand.find_by(offence_category: offence_category, description: self['band'])
+  end
+
+  def contrary_to
+    self['contrary_to']
+  end
+
+  def year_chapter
+    self['year_chapter']
+  end
+end
+
+class CSV::Row
```

Comment:
**[not a blocker]** Even though in this case, this patching seems quite self contained, keep seeing core ruby classes/gem classes patched makes me really 😨 . It's **not** a good practice and its only prone to come and bites us in the future if those core ruby classes/gem classes change their interface.

Personally, to achieve that same behaviour, I would just pass the row to a `OffenceCSVRow` instance and call `to_h` on it to give me the key/value pairs that you need to create the offence in the database.

Reply:
the only extension to `CSV::Row` it is performing is calling the hash directly which id be surprised/worried if they changed in the core class. The idea is simply to neaten up the seed file which was/is complex enough. 

i guess your idea sacrifices brevity for not patching